lib/valkyrie/storage/fedora.rb

```diff
@@ -106,7 +106,7 @@ def upload_file(fedora_uri:, io:, content_type: "application/octet-stream", orig
       connection.http.put do |request|
         request.url fedora_uri
         request.headers['Content-Type'] = content_type
-        request.headers['Content-Length'] = io.length.to_s if io.respond_to?(:length)
+        request.headers['Content-Length'] = (io.length.to_s if io.respond_to?(:length)) || (io.size.to_s if io.respond_to?(:size))
```

Comment:
if it responds to neither, will it set the content-length header to `nil`, or not set it at all?

Reply:
Good point.